plugins/catalog-backend/migrations/20220616202842_refresh_keys.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+exports.up = async function up(knex) {
+  await knex.schema.createTable('refresh_keys', table => {
+    table.comment(
+      'This table contains relations between entities and keys to trigger refreshes with',
+    );
+    table
+      .text('entity_ref')
```

Comment:
Let's make the foreign key constraint on the entity_id in the refresh_state table and also add a cascading delete so that we clear items in this table when we drop from refresh_state.

See https://github.com/backstage/backstage/blob/7187adfadd7d06498e015f3ac26eaaf5ebc8c01d/plugins/catalog-backend/migrations/20210302150147_refresh_state.js#L152 for inspiration

Reply:
@kissmikijr did you tun into trouble using and referencing `entity_id` here instead? It's the PK so a bit nicer to use that rather than the ref. I believe it should be available when needed as well.

Reply:
Nope, I actually thought we will need the entity_ref, but just realized that the id is more than enough since we are going to refresh based on the key anyways.